dandischema/metadata.py

```diff
@@ -265,45 +265,124 @@ def validate(
+    except ValueError as e:
+        msg = (
+            "The provided target version, {to_version!r}, is not a valid DANDI schema "
+            "version."
         )
+        raise ValueError(msg) from e
+
+    # Permit only allowed target schemas
     if to_version not in ALLOWED_TARGET_SCHEMAS:
-        raise ValueError(f"Current target schemas: {ALLOWED_TARGET_SCHEMAS}.")
-    schema_version = obj.get("schemaVersion")
-    if schema_version == DANDI_SCHEMA_VERSION:
-        return obj
-    if schema_version not in ALLOWED_INPUT_SCHEMAS:
-        raise ValueError(f"Current input schemas supported: {ALLOWED_INPUT_SCHEMAS}.")
-    if version2tuple(schema_version) > version2tuple(to_version):
-        raise ValueError(f"Cannot migrate from {schema_version} to lower {to_version}.")
-    if not (skip_validation):
-        schema = _get_schema(schema_version, "dandiset.json")
+        msg = (
+            f"Target version, {to_version!r}, is not a supported target schemas, "
+            f"{ALLOWED_TARGET_SCHEMAS}"
+        )
+        raise ValueError(msg)
+    # ----------------------------------------------------------------
+
+    # Ensure the target DANDI schema version is at least the DANDI schema version
+    # of the provided instance
+    if instance_ver_tuple > target_ver_tuple:
+        raise ValueError(f"Cannot migrate from {instance_ver} to lower {to_version}.")
+
+    # Optionally validate the instance against the DANDI schema it specifies
+    # before migration
+    if not skip_validation:
+        schema = _get_schema(instance_ver, "dandiset.json")
         _validate_obj_json(obj, schema)
-    if version2tuple(schema_version) < version2tuple("0.6.0"):
-        for val in obj.get("about", []):
+
+    instance_migrated = deepcopy(obj)
+
+    if instance_ver_tuple == target_ver_tuple:
+        return instance_migrated
+
+    if instance_ver_tuple < version2tuple("0.6.0") <= target_ver_tuple:
+        for val in instance_migrated.get("about", []):
             if "schemaKey" not in val:
                 if "identifier" in val and "UBERON" in val["identifier"]:
                     val["schemaKey"] = "Anatomy"
                 else:
                     raise ValueError("Cannot auto migrate. SchemaKey missing")
-        for val in obj.get("access", []):
+        for val in instance_migrated.get("access", []):
             if "schemaKey" not in val:
                 val["schemaKey"] = "AccessRequirements"
-        for resource in obj.get("relatedResource", []):
+        for resource in instance_migrated.get("relatedResource", []):
             resource["schemaKey"] = "Resource"
-        if "schemaKey" not in obj["assetsSummary"]:
-            obj["assetsSummary"]["schemaKey"] = "AssetsSummary"
-        if "schemaKey" not in obj:
-            obj["schemaKey"] = "Dandiset"
-        obj["schemaVersion"] = to_version
-    return obj
+        if "schemaKey" not in instance_migrated["assetsSummary"]:
```

Comment:
```suggestion
        if "schemaKey" not in instance_migrated.get("assetsSummary", {}):
```

Reply:
I replaced the line with the following instead.

```python
        if (
            "assetsSummary" in obj_migrated
            and "schemaKey" not in obj_migrated["assetsSummary"]
        ):
```